### PR TITLE
[3.x] Add autocompletion for `AnimatedSprite.play()`

### PR DESCRIPTION
**Pull request by timothyqiu**
`3.x` version of #60756

### CODE REVIEW

File: scene/2d/animated_sprite.cpp [hunk 1]
```diff
@@ -74,6 +78,18 @@ bool AnimatedSprite::_edit_use_rect() const {
 	}
 	return t.is_valid();
 }
+
+void AnimatedSprite::get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const {
+	if (p_idx == 0 && p_function == "play" && frames.is_valid()) {
+		const String quote_style = EDITOR_GET("text_editor/completion/use_single_quotes") ? "'" : "\"";
+		List<StringName> al;
```
Comment (lawnjelly) [inline scene/2d/animated_sprite.cpp:85]: I'm not very familiar with this but I noticed some of the other places in 3.x that do this, use this code:
```
#ifdef TOOLS_ENABLED
	const String quote_style = EDITOR_GET("text_editor/completion/use_single_quotes") ? "'" : "\"";
#else
	const String quote_style = "\"";
#endif
```
Is this necessary or does the EDITOR_GET work ok in non-tools builds?

Comment (timothyqiu) [reply scene/2d/animated_sprite.cpp:85]: Ah, this whole `get_argument_options()` function is inside `TOOLS_ENABLED`, so it's not necessary to use the `#ifdef`.

I'm not sure if it's best practice. But I'm going to change it to use the `#ifdef` approach for the sake of consistency.

**Review by lawnjelly — APPROVED** [review]
I'm not an expert on this area but it looks fine to me.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**